script.js

```diff
@@ -1,12 +1,440 @@
+const showSelectHtml = document.querySelector("#show-select-html");
+const showListingButton = document.querySelector("#show-listing-button");
+
+// Global variables
+let allEpisodes;
+let allShows;
+
+window.onload = setup;
+
+// Fetch episodes function
 function setup() {
-  const allEpisodes = getAllEpisodes();
-  makePageForEpisodes(allEpisodes);
+  allShows = getAllShows().sort((a, b) => {
+    return a.name.localeCompare(b.name); // sort shows into alphabeltical order by comparing string, not numbers
+  });
+
+  buildShowDropdownList(allShows);
+  makePageForShows(allShows);
+}
+
+function fetchShowEpisodes(showId) {
+  fetch(`https://api.tvmaze.com/shows/${showId}/episodes`) // returns a promise and it is pending
+    .then((response) => {
+      if (response.status >= 200 && response.status <= 299) {
+        return response.json(); // returns the promise as fulfilled and gives us a response object which we pass into the callback function
+      } else {
+        throw new Error(
+          `Encountered something unexpected: ${response.status} ${response.statusText}`
+        );
+      }
+    })
+    .then((data) => {
+      allEpisodes = data;
+      showSearchInputHtml.value = "";
+      makePageForEpisodes(allEpisodes);
+      buildEpisodeDropdownList(allEpisodes);
+    })
+    .catch((error) => {
+      console.log(error);
+    });
 }
 
+// level 100
+
+/*
+  If the user clicks show select then the show select will disappear
+  If the user clicks the see show listing button then the episodes will disappear and the shows will appear also the episode list should not be shown initially
+*/
+
 function makePageForEpisodes(episodeList) {
-  const rootElem = document.getElementById("root");
-  rootElem.textContent = `Got ${episodeList.length} episode(s)`;
+  episodeSelectHtml.style.display = "inline"; // make the episode dropdown list visible
+  showSelectHtml.style.display = "none"; // hide the episode dropdown list
+
+  episodeSearchInputHtml.style.display = "inline"; // make the show searchbar visible
+  showSearchInputHtml.style.display = "none"; // hide episode searchbar from show page
+
+  episodeSearchResultDisplayHtml.style.display = "inline";
+  showSearchResultDisplayHtml.style.display = "none"; // hide episode search result from show page
+
+  showSearchInputHtml.removeEventListener("input", searchForShow);
+  episodeSearchInputHtml.addEventListener("input", searchForEpisode);
+
+  console.log(showSelectHtml, "<---ShowSelectHTML");
+  rootHtml.innerHTML = "";
+  const cardsContainerHtml = document.createElement("div"); // Creating the cards container element inside the rootEl
+
+  cardsContainerHtml.classList.add("episode-cards-container");
+
+  // Loop through each episode and create a card for it
+
+  episodeList.forEach((episode, index) => {
+    // Get the episode name, season number, episode number, and combine them
+    const episodeName = episode.name;
+    const paddedSeasonNum = ("0" + episode.season).slice(-2);
```

Comment:
Rather than repeating this code on this and the next line, maybe we could extract a function with a clear name which shows what it's doing?

Reply:
Thank you for pointing this out! I created 2 different functions to format the name in the episode dropdown list and to format episode titles in episode cards. I had to create two functions because they are formatted slightly differently.